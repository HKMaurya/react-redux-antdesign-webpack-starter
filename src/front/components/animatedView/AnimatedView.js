// @flow

// #region imports
import React, { Component } from 'react';
import cx from 'classnames';
// #endregion

// #region flow types
export type Props = {
  children: React.ReactNode,
  animated: boolean,
  ...any,
};

export type State = { ...any };
// #endregion

class AnimatedView extends Component<Props, State> {
  static defaultProps = {
    animated: true,
  };

  // #region lifecycle
  render() {
    const { animated, children } = this.props;

    return (
      <section
        className={cx({
          content: true,
          'view-enter': animated,
        })}
      >
        {children}
      </section>
    );
  }
  // #endregion
}

export default AnimatedView;
