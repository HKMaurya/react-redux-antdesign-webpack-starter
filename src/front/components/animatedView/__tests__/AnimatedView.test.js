// @flow

// #region imports
import React from 'react';
import renderer from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme
import AnimatedView from '../AnimatedView';
// #endregion

describe('AnimatedView component', () => {
  it('renders as expected', () => {
    const props = {
      minScrollY: 10,
      scrollTo: null,
      onScrollDone: () => {},
    };

    const component = renderer
      .create(
        <div>
          <AnimatedView {...props}>
            <p>a child</p>
          </AnimatedView>
        </div>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
