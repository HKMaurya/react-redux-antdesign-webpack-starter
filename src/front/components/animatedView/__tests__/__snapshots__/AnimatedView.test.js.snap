// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedView component renders as expected 1`] = `
<div>
  <section
    className="content view-enter"
  >
    <p>
      a child
    </p>
  </section>
</div>
`;
